### PR TITLE
Remove localforage and point to upstream typings instead

### PR DESCRIPTION
**Pull request by jmalonzo**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `tsc` without errors.
- [ ] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by jmalonzo** [comment]
Not sure how to make the test pass atm. Appreciate any hints if it needs fixing prior to merging.

File: localforage/package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (mhegazy) [inline localforage/package.json]: you need to remove the whole folder.

Comment (jmalonzo) [inline localforage/package.json]: Thanks. I followed the README so it points to the upstream package with the bundled types. Is that OK or should I totally remove the folder altogether?

Comment (mhegazy) [inline localforage/package.json]: yes. the whole folder should be deleted.

**Comment by jmalonzo** [comment]
Thanks @mhegazy. Done.

**Comment by paulvanbrenk** [comment]
Manually merged

**Pull request closed without merging**